packages/vite-plugin/src/index.ts

```diff
@@ -10,53 +9,99 @@ import {
   getPackageInfo,
   CompileOptions,
   transform,
+  type Compiler,
+  createCompiler,
+  normalizePath,
 } from '@vanilla-extract/integration';
-import { type PostCSSConfigResult, resolvePostcssConfig } from './postcss';
-
-const styleUpdateEvent = (fileId: string) =>
-  `vanilla-extract-style-update:${fileId}`;
 
 const virtualExtCss = '.vanilla.css';
-const virtualExtJs = '.vanilla.js';
+
+const fileIdToVirtualId = (id: string) => `${id}${virtualExtCss}`;
+const virtualIdToFileId = (virtualId: string) =>
+  virtualId.slice(0, -virtualExtCss.length);
 
 interface Options {
   identifiers?: IdentifierOption;
-  emitCssInSsr?: boolean;
   esbuildOptions?: CompileOptions['esbuildOptions'];
+  mode?: 'transform' | 'emitCss';
```

Comment:
The presence of `esbuildOptions` will switch to the esbuild-based compiler. And in `mode: 'transform'` we don't need the new compiler.